test_workspace/ApplicativeTraverse.bosatsu

```diff
@@ -0,0 +1,38 @@
+package Bosatsu/Example/ApplicativeTraverse
+
+export [Applicative(), Traverse(), traverse_List]
+
+# Represents the Applicative typeclass
+struct Applicative(
+  pure: forall a. a -> f[a],
+  map: forall a, b. (a -> b) -> f[a] -> f[b],
+  product: forall a, b. f[a] -> f[b] -> f[(a, b)])
+
+# Represents the Traverse typeclass
+struct Traverse[g](
```

Comment:
either don't put `g` here or put `f` above. No reason to put g here really, it is the only free type parameter.